From: Alan Jenkins <alan.christopher.jenkins@gmail.com>
Date: Sat, 3 Mar 2018 18:44:39 +0000
Subject: [PATCH 14/19] login: remember that fds received from PID1 need to be
 removed eventually

Remember to set sd->pushed_fd when we receive an fd from PID1 on startup,
the same as we set it when we send an fd to PID1.

(cherry picked from commit f8f9419e8765dd427301fa479fb014c5f92250ed)
---
 src/login/logind-session-device.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/login/logind-session-device.c b/src/login/logind-session-device.c
index 7253f95..73eee72 100644
--- a/src/login/logind-session-device.c
+++ b/src/login/logind-session-device.c
@@ -543,5 +543,6 @@ void session_device_attach_fd(SessionDevice *sd, int fd, bool active) {
         assert(!sd->active);
 
         sd->fd = fd;
+        sd->pushed_fd = true;
         sd->active = active;
 }
